platforms/platforms.go

```diff
@@ -59,7 +62,11 @@ func DetermineBazelFilename(version string, includeSuffix bool) (string, error)
 		filenameSuffix = DetermineExecutableFilenameSuffix()
 	}
 
-	return fmt.Sprintf("bazel-%s-%s-%s%s", version, osName, machineName, filenameSuffix), nil
+	bazelFlavor := "bazel"
+	if configs.GetEnvOrConfig("BAZELISK_NOJDK") != "0" && configs.GetEnvOrConfig("BAZELISK_NOJDK") != "" {
```

Comment:
why don't we just set the convention to that `BAZELISK_NOJDK=1` is accepted, and everything else is rejected?

That would make this a lot easier to use/manage vs having to read the code to find these 2 magic values to avoid.

If we want to stay consistent with the python implementation then we would want GetEnvOrConfig here to accept a default return value in case both env and config were not set?